modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+        Assert.assertEquals(returnedURI, expectedURI);
+
+        if (attributes == null & excludeAttributes == null) {
+
+            Assert.assertTrue(obj.has("emails"));
+
+        } else if (attributes == null & excludeAttributes != null) {
+
+            Assert.assertFalse(obj.has("emails"));
+
+        }
+
+    }
+
+    @DataProvider(name = "dataForGetNotFoundException")
+    public Object[][] dataToGetNotFoundException() {
+
+        return new Object[][]{
+                {"David", "userName", null, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetNotFoundException")
+    public void testGetUserNotFoundException(String name, String attributes, String excludeAttributes,
+                                             int expectedScimResponseStatus)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForGetCharonException")
+    public Object[][] dataToGetCharonException() {
+
+        return new Object[][]{
+                {"David", "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetCharonException")
+    public void testGetUserCharonException(String name, String attributes, String excludeAttributes,
+                                           int expectedScimResponseStatus)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
```

Comment:
can't we merge this into 1 line?

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d